### PR TITLE
Replace aggregatable source registration JSON list with dict

### LINKED ISSUE #431: Consider replacing Attribution-Reporting-Register-Aggregatable-Source JSON list with dictionary
Since the IDs are required to be unique anyway, this will simplify the JSON structure.

Before:
```json
[
  {"id": "campaignCounts", "key_piece": "0x159"},
  {"id": "geoValue", "key_piece": "0x5"}
]
```

After:
```json
{
  "campaignCounts": "0x159",
  "geoValue": "0x5"
}
```

https://github.com/WICG/conversion-measurement-api/blob/main/AGGREGATE.md#attribution-source-registration

### PR DESCRIPTION
**Pull request by linnan-github**
Fixes #431

### CODE REVIEW

File: AGGREGATE.md [hunk 1]
```diff
@@ -72,22 +72,18 @@ example an ad-tech will use the API to collect:
 Registering sources eligible for aggregate reporting entails adding a new header
 on the response to the `attributionsrc` request:
 `Attribution-Reporting-Register-Aggregatable-Source`, in the form of a JSON
-list.
+dict.
```
Comment (apasel422) [inline AGGREGATE.md:75]:
```suggestion
dictionary.
```

File: AGGREGATE.md [hunk 2]
```diff
@@ -72,22 +72,18 @@ example an ad-tech will use the API to collect:
...
+  // Generates a "0x159" key piece (low order bits of the key) for the key named
+  // "campaignCounts".
+  "campaignCounts": "0x159", // User saw ad from campaign 345 (out of 511)
+
+  // Generates a "0x5" key piece (low order bits of the key) for the key named "geoValue".
+  "geoValue": "0x5" // Source-side geo region = 5 (US), out of a possible ~100 regions
+
 ```
-This defines a list named histogram contributions, each with a piece of the
+This defines a dict of named aggregation keys, each with a piece of the
```
Comment (apasel422) [inline AGGREGATE.md:86]:
```suggestion
This defines a dictionary of named aggregation keys, each with a piece of the
```

File: AGGREGATE.md [hunk 3]
```diff
@@ -72,22 +72,18 @@ example an ad-tech will use the API to collect:
...
-},
 {
-  // Generates a "0x5" key piece (low order bits of the key) for the key named "geoValue"
-  "id": "geoValue",
-  // Source-side geo region = 5 (US), out of a possible ~100 regions.
-  "key_piece": "0x5"
-}]
+  // Generates a "0x159" key piece (low order bits of the key) for the key named
+  // "campaignCounts".
+  "campaignCounts": "0x159", // User saw ad from campaign 345 (out of 511)
```
Comment (apasel422) [inline AGGREGATE.md:80]: Did we decide whether to use a string here or a dictionary to support future changes?

CC @johnivdel @csharrison

Comment (johnivdel) [reply AGGREGATE.md:80]: Updated the issue, let's go with a string to keep that schema as simple as possible.

Comment (linnan-github) [reply AGGREGATE.md:80]: SGTM, thanks!

**Review by apasel422 — APPROVED** [review]

**Review by johnivdel — APPROVED** [review]

**Pull request merged**